src/julia-syntax.scm

```diff
@@ -2112,9 +2112,17 @@
                                   x (make-ssavalue)))
                          (ini (if (eq? x xx) '() (list (sink-assignment xx (expand-forms x)))))
                          (n   (length lhss))
+                         ;; skip last assignment if it is an all-underscore vararg
+                         (n   (if (> n 0)
+                                  (let ((l (last lhss)))
+                                    (if (and (pair? l) (eq? (car l) '|...|)
+                                             (underscore-symbol? (cadr l)))
+                                        (- n 1)
+                                        n))
+                                  n))
                          (st  (gensy)))
                     `(block
-                      (local ,st)
+                      ,.(if (> n 0) `((local ,st)) '())
```

Comment:
`() = a` actually currently produces a dangling `NewVarNode`, which I guess is not a big deal, especially since it's a pretty contrived example, but I guess avoiding this doesn't hurt?

Reply:
Yeah it's worth removing the unused NewvarNode.